From 6cbe3e6e27f98c728e9d2bf1f8fa8b69702a03bc Mon Sep 17 00:00:00 2001
From: Varsha Maira <vmaira@habana.ai>
Date: Wed, 16 Apr 2025 13:53:17 +0300
Subject: [PATCH] [SW-218933] g3: use access mode enum from specs

The `mem_access_mode` enum was defined within individual Zephyr apps. This
enum has been moved to specs dir to avoid definitions across multiple apps
in an earlier patch. This patch replaces the older enum with enum from
specs dir in baremetal preboot.

Change-Id: Ie6eff8e564e4a7eab304cfb16bff3707f9f1884f
Signed-off-by: Varsha Maira <vmaira@habana.ai>
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 87d7d3f..98abe9c 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -1138,7 +1138,7 @@
 }
 
 void preboot_security_dynamic_img_sram_open(enum mem_region img_region,
-				enum preboot_mem_access_mode mode)
+				enum mem_access_mode mode)
 {
 	struct hl_sec_dyn_pcie_rr_cfg *cfg;
 	struct mem_desc mdesc;
@@ -1177,15 +1177,15 @@
 	/* Finally open the image memory space */
 	cfg = &preboot_sec_dyn_cfg[IMAGE];
 	switch (mode) {
-	case MEM_ACCESS_RO:
+	case ACCESS_RO:
 		/* Open IMAGE region in read only mode */
 		hl_secure_dyn_rr_cfg(HL_SEC_HBW_SRAM_RO, cfg->min, cfg->max);
 		break;
-	case MEM_ACCESS_RW:
+	case ACCESS_RW:
 		/* Open IMAGE region read-only RR if closed already */
 		hl_secure_dyn_rr_cfg(HL_SEC_HBW_SRAM_RO, 0, 0);
 		break;
-	case MEM_ACCESS_WR: /* write-only not possible in gaudi2 */
+	case ACCESS_WR: /* write-only not possible in gaudi2 */
 	default:
 		preboot_puts_hex("dynamic_image_sram_open with invalid mode: ", mode, false);
 		return;
diff --git a/arch/gaudi3/common/security.c b/arch/gaudi3/common/security.c
index 1edb405..5d5bc59 100644
--- a/arch/gaudi3/common/security.c
+++ b/arch/gaudi3/common/security.c
@@ -1064,7 +1064,7 @@
 
 #define IMAGE_OFFSET_MASK	0xFFFFFF
 void preboot_security_dynamic_img_sram_open(enum mem_region img_region,
-				enum preboot_mem_access_mode mode)
+				enum mem_access_mode mode)
 {
 	struct mem_desc mdesc;
 	uint64_t img_max, image_base_full;
@@ -1104,7 +1104,7 @@
 	 */
 	hl_sec_config_dyn_hbw_rr(HL_SEC_HBW_RR_7, img_max, RR_HBW_PRIV_END_RSVD, HL_SEC_RD_WR_PROT);
 
-	if (mode == MEM_ACCESS_RO)
+	if (mode == ACCESS_RO)
 		hl_sec_config_dyn_hbw_rr(HL_SEC_HBW_RR_5, image_base_full, img_max, HL_SEC_WR_PROT);
 
 	/*
diff --git a/include/security.h b/include/security.h
index a87c845..4b31058 100644
--- a/include/security.h
+++ b/include/security.h
@@ -9,12 +9,7 @@
 
 #include <types.h>
 #include <memory.h>
-
-enum preboot_mem_access_mode {
-	MEM_ACCESS_RO,
-	MEM_ACCESS_WR,
-	MEM_ACCESS_RW,
-};
+#include <hl_sec_if.h>
 
 bool preboot_security_is_real_efuse_sec_en(void);
 bool preboot_security_is_enabled(void);
@@ -24,7 +19,7 @@
 void preboot_security_dynamic_vers_init(u64 ver_off, u32 size);
 void preboot_security_dynamic_img_sram_close(void);
 void preboot_security_dynamic_img_sram_open(enum mem_region img_region,
-					    enum preboot_mem_access_mode mode);
+					    enum mem_access_mode mode);
 void preboot_security_dynamic_img_elbi_reg_close(void);
 void preboot_security_dynamic_img_elbi_reg_open(void);
 void preboot_security_cleanup_before_run(void);
diff --git a/lib/preboot/module_fwupd.c b/lib/preboot/module_fwupd.c
index 744fa20..68973e8 100644
--- a/lib/preboot/module_fwupd.c
+++ b/lib/preboot/module_fwupd.c
@@ -15,6 +15,7 @@
 #include <hl_fw_if.h>
 #include <hl_fw_upd_if.h>
 #include <hl_fit_if.h>
+#include <hl_sec_if.h>
 #include <fit_load_type.h>
 #include <security.h>
 #include <memory.h>
@@ -69,7 +70,7 @@
 __weak void preboot_security_dynamic_img_elbi_reg_open(void) {}
 __weak void preboot_security_dynamic_img_elbi_reg_close(void) {}
 __weak void preboot_security_dynamic_img_sram_open(enum mem_region img_region,
-					enum preboot_mem_access_mode mode)
+					enum mem_access_mode mode)
 {
 }
 
@@ -210,7 +211,7 @@
 		 */
 		if (state->use_security && status.offset &&
 		    state->wfc_cmd.cmd != HL_UPD_MAP_SRAM)
-			preboot_security_dynamic_img_sram_open(state->mregion, MEM_ACCESS_RO);
+			preboot_security_dynamic_img_sram_open(state->mregion, ACCESS_RO);
 	}
 
 	state->sts_writel(status.val);
@@ -506,7 +507,7 @@
 	if (state->use_security) {
 		if (cmd == HL_UPD_MAP_SRAM)
 			preboot_security_dynamic_img_sram_open(state->mregion,
-				MEM_ACCESS_RW);
+							       ACCESS_RW);
 		else
 			preboot_security_dynamic_img_elbi_reg_open();
 	}
